walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,94 @@
+    // If there are no attendees then return WHOLE_DAY.
+    if (request.getAttendees().isEmpty() || request.getDuration() == 0){
+      return solutions;
+    }
+    for (Event event: events){
+        // Check if event attendees are in the request. If not continue to next event.
+        boolean relevantEvent = overlappingAttendees(request, event);
+        if (!relevantEvent){
+            continue;
+        }
+        // Take out conflicting time from current answer of TimeRanges.
+        solutions = takeOutConflicts(solutions, event, request.getDuration());
+    }
+    return solutions;
+    //cannot conflict with the collection of events given.
+    //make sure to check end point of time range is inclusive. 
+  }
+
+  /**Splits a TimeRange into two time or shotens one TimeRange due to conflict with a given event */
+  private ArrayList<TimeRange> takeOutConflicts(ArrayList<TimeRange> oldSolutions, Event event, long duration) {
+      ArrayList<TimeRange> solutions = new ArrayList<>();
+      for (TimeRange slot: oldSolutions){
+          if (slot.contains(event.getWhen())){
+              //need to split slot into two
+              TimeRange slotA = TimeRange.fromStartEnd(slot.start(), event.getWhen().start(), false);
+              TimeRange slotB = TimeRange.fromStartEnd(event.getWhen().end(), slot.end(), false);
+              //check if duration is still good for both
+              if (durationCheck(duration, slotA)){
+                  solutions.add(slotA);
+              }
+              if (durationCheck(duration, slotB)){
+                  solutions.add(slotB);
+              }
+          }
+          else if (slot.overlaps(event.getWhen())){
+              // if event contains slot continue in the loop
+              if (event.getWhen().contains(slot)){
+                  continue;
+              }
+              //shorten slot in the front
+              else if (event.getWhen().end() > slot.start()){
+                   TimeRange shortenedSlot = TimeRange.fromStartEnd(event.getWhen().end(), slot.end(), false);
+                   if (durationCheck(duration, shortenedSlot)){
+                       solutions.add(shortenedSlot);
+                   }
+              }
+              else{
+                  TimeRange shortenedSlot = TimeRange.fromStartEnd(slot.start(), event.getWhen().start(), false);
+                  if (durationCheck(duration, shortenedSlot)){
+                      solutions.add(shortenedSlot);
+                  }
+              }
+              
+          }
+          else{
+              solutions.add(slot);
+          }
+      }
+      return solutions;
+    
+  }
+
+  /**Returns true if at least one attendee in Meeting Request is an attendee in the Event */
+  private boolean overlappingAttendees(MeetingRequest request, Event event){
+      for (String attendee : event.getAttendees()){
+          if (request.getAttendees().contains(attendee))
+            return true;
+        }
+      return false;
```

Comment:
You can reduce this to a single operation using [`Collection#retainAll`](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#retainAll-java.util.Collection-):

```java
return !request.getAttendees().retainAll(event.getAttendees()).isEmpty();
```